core/src/main/java/nl/nn/adapterframework/http/WebServiceSender.java

```diff
@@ -116,28 +117,32 @@ protected HttpRequestBase getMethod(URI uri, String message, ParameterValueList
 		}
 
 		String soapmsg;
-		if (isSoap()) {
-			soapmsg = soapWrapper.putInEnvelope(message, getEncodingStyle(), serviceNamespaceURI, null, getNamespaceDefs());
-		} else {
-			soapmsg = message;
+		try {
+			if (isSoap()) {
+				soapmsg = soapWrapper.putInEnvelope(message.asString(), getEncodingStyle(), serviceNamespaceURI, null, getNamespaceDefs());
+			} else {
+				soapmsg = message.asString();
+			}
+		} catch (IOException e) {
+			throw new SenderException(getLogPrefix()+"error reading message", e);
 		}
 
 		if (wsscf!=null) {
 			soapmsg = soapWrapper.signMessage(soapmsg, wsscf.getUsername(), wsscf.getPassword(), isWssPasswordDigest());
 		}
 		if (log.isDebugEnabled()) log.debug(getLogPrefix()+"SOAPMSG [" + soapmsg + "]");
 
-		HttpRequestBase method = super.getMethod(uri, soapmsg, parameters, session);
+		HttpRequestBase method = super.getMethod(uri, Message.asMessage(soapmsg), parameters, session);
```

Comment:
Ik zou hier new Message(soapmsg) gebruiken, omdat je weet dat het een string is.
Dan weet je ook dat je een waarschuwing krijgt als je ooit van soapmsg een Message maakt, en het dan dus opruimt.